### PR TITLE
[WIP] Improve robustness of krunner handling in wayland scenarios - more aggressive

### PR DESCRIPTION
**Pull request by okurz**
Verification runs:
See https://progress.opensuse.org/issues/35589#note-85 for details of verification runs

Related progress issue: https://progress.opensuse.org/issues/35589

### CODE REVIEW

**Comment by okurz** [comment]
@Vogtinator WDYT?

File: lib/opensusebasetest.pm [hunk 1]
```diff
@@ -376,24 +376,24 @@ sub select_bootmenu_more {
...
-            upload_logs $fn;
-        }
-        else {
-            my $fn  = '/tmp/kde4_configs.tar.bz2';
-            my $cmd = sprintf 'tar cjf %s /home/%s/.kde4/share/config/*rc', $fn, $username;
-            type_string "$cmd\n";
-            upload_logs $fn;
-        }
-        save_screenshot;
+    return unless check_var('DESKTOP', 'kde');
```
Comment (Vogtinator) [inline lib/opensusebasetest.pm]: Shouldn't this be before `save_screenshot;`

File: tests/x11/krunner.pm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+use warnings;
+use testapi;
+use x11utils 'desktop_runner_hotkey';
+
+sub run {
+    wait_screen_change { send_key desktop_runner_hotkey };
+    type_string 'echo "Hello World"';
+    assert_screen 'krunner-hello_world';
+    send_key 'ret';
+    # on wayland krunner sometimes vanishes or crashes, e.g. after the second
```
Comment (Vogtinator) [inline tests/x11/krunner.pm]: I've never seen that, at least not outside of openQA.

File: tests/x11/reboot_plasma5.pm [hunk 3]
```diff
@@ -22,9 +22,6 @@ sub run {
     send_key "ctrl-alt-delete";    # reboot
     assert_and_click 'sddm_reboot_option_btn';
     $self->wait_boot(bootloader_time => 300);
-    # Ensure the desktop runner is reactive again before going into other test
-    # modules
-    # https://progress.opensuse.org/issues/30805
     $self->check_desktop_runner;
```
Comment (Vogtinator) [inline tests/x11/reboot_plasma5.pm]: IMO a a comment here should stay.

File: lib/opensusebasetest.pm (hunk 1 above)
Comment (okurz) [reply lib/opensusebasetest.pm]: true however I simply have not changed the location. But with removal of the `select_console` call I should probably just do that.

**Comment by Vogtinator** [comment]
On first use, krunner loads plugins only after typing, so maybe it's enough to type a single letter, wait a second and then type the rest quickly?

IMO disabling plugins should be avoided unless absolutely necessary (but it looks like it almost is that bad).

File: tests/x11/krunner.pm (hunk 2 above)
Comment (okurz) [reply tests/x11/krunner.pm]: And that is basically the whole point of this PR and the according ticket https://progress.opensuse.org/issues/35589 :)

https://openqa.opensuse.org/tests/871386/file/video.ogv#t=32.08,32.14 shows a good example. Play it and you see that the first characters are correctly typed, then suddenly the desktop runner vanishes and reappears but without the previously typed characters.

File: lib/susedistribution.pm [hunk 4]
```diff
@@ -161,9 +161,9 @@ sub init_desktop_runner {
     # https://progress.opensuse.org/issues/35589
     if (check_var('DESKTOP', 'kde')) {
         if (get_var('WAYLAND')) {
-            wait_still_screen(3);
-            type_string_very_slow substr $program, 0, 2;
-            wait_still_screen(3);
+            wait_still_screen(2);
+            type_string_very_slow substr $program, 0, 1;
+            type_string_very_slow substr $program, 1, 1;
             type_string_very_slow substr $program, 2;
```
Comment (okurz) [inline lib/susedistribution.pm]: @Vogtinator the "typing character by character" part is already in here. I will try to push it further by checking if the desktop runner is there after every single character on wayland.

File: tests/x11/krunner.pm (hunk 2 above)
Comment (Vogtinator) [reply tests/x11/krunner.pm]: It's neither a crash nor a relayout (which happens as the window size changes). I'll try to reproduce that somehow (but already tried multiple times, without "success").

File: lib/susedistribution.pm (hunk 4 above)
Comment (Vogtinator) [reply lib/susedistribution.pm]: I don't think that would mitigate the issue - it would only detect the failure earlier.

**Comment by ggardet** [comment]
We already have `desktop_runner` test. Maybe you could integrate your changes inside?

**Comment by okurz** [comment]
with recent submissions touching krunner it seems the stability of krunner has increased and we do not need this in the current way anymore

**Pull request closed without merging**